### PR TITLE
HOTFIX: check logic of KAFKA-2515 should be on buffer.limit()

### PR DESCRIPTION
**Pull request by guozhangwang**

### CODE REVIEW

**Comment by guozhangwang** [comment]
Ping @junrao @ewencp for reviews.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java [hunk 1]
```diff
@@ -512,8 +512,8 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                         this.subscriptions.fetched(tp, record.offset() + 1);
                         this.records.add(new PartitionRecords<>(fetchOffset, tp, parsed));
                         this.sensors.recordsFetchLag.record(partition.highWatermark - record.offset());
-                    } else if (buffer.capacity() >= this.fetchSize) {
-                        // we did not read a single message from a max fetchable buffer
+                    } else if (buffer.limit() > 0) {
+                        // we did not read a single message from an non-empty buffer
```
Comment (junrao) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java]: an non-empty => a non-empty

**Comment by junrao** [comment]
Thanks for the patch. Just a minor comment. Otherwise, LGTM.

**Comment by ewencp** [comment]
Agreed, LGTM and validated that this was (at least one case...) causing the system tests to fail.

**Pull request closed without merging**